src/tiled/mainwindow.cpp

```diff
@@ -816,6 +817,27 @@ bool MainWindow::saveFileAs()
     return saveFile(fileName);
 }
 
+void MainWindow::saveAll()
+{
+    foreach (MapDocument *mapDoc, mDocumentManager->documents()) {
+        QString fileName(mapDoc->fileName());
+        bool successfullSave = true;
+        QString error;
+        if (fileName.isEmpty()) {
+            successfullSave = saveFileAs();
+            error = tr("Target filename is empty.");
```

Comment:
This error makes little sense. When the `saveFileAs` does not return true, the user already got an appropriate error message or pressed cancel, in which case the `saveAll` action should just abort.